components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIEndpointUrlExtractorUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.impl.utils;
+
+import org.wso2.carbon.apimgt.api.APIConsumer;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIManagerFactory;
+import org.wso2.carbon.context.CarbonContext;
+
+/**
+ * Utility class for API endpoint URL extractor.
+ */
+public class APIEndpointUrlExtractorUtils {
+
+    /**
+     * Gets the APIConsumer which is corresponding to the current logged in user taken from the carbon context
+     *
+     * @return an APIConsumer which is corresponding to the current logged in user
+     * @throws APIManagementException
+     */
+    public static APIConsumer getLoggedInUserConsumer() throws APIManagementException {
+        return APIManagerFactory.getInstance().getAPIConsumer(getLoggedInUsername());
+    }
+
+    /**
+     * Gets the current logged in user taken from the carbon context
+     *
+     * @return the current logged in user taken from the carbon context
+     */
+    public static String getLoggedInUsername() {
+        return CarbonContext.getThreadLocalCarbonContext().getUsername();
+    }
```

Comment:
Shall we move these methods to APIManagerFactory itself? Then we can easily reuse them in other cases too.
Make getLoggedInUsername() method private